### PR TITLE
[RISCV] Combine vslideup_vl with known VL to a smaller LMUL

### PR DESCRIPTION
**Pull request by lukel97**
Similiar to #66267, we can perform a vslideup_vl on a smaller type if we know the highest lane that will be written to, which can be determined from VL.

This is an alternative to #65997 and #66087

Stacked upon #66267

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rvv/fixed-vectors-int-explodevector.ll
Comment (preames) [inline llvm/test/CodeGen/RISCV/rvv/fixed-vectors-int-explodevector.ll:814]: This looks like the consequence of the vslidedown combine, but in a vslideup change.  I think maybe you got the rebase wrong?

**Review by preames — CHANGES_REQUESTED** [review]
(On hold until slidedown questions resolved)

**Comment by lukel97** [comment]
Marking this as a draft as it needs rebased after #66087 and #65997 have landed. It also remains to be seen if the combine approach will be able to fully subsume performing the reduction in lowering.

**Pull request closed without merging**
